lib/tools/ui/widgets/text_entry.dart

```diff
@@ -77,26 +77,33 @@ class TextEntry extends StatelessWidget {
         ),
       ),
       validator: (value) {
-        if (canBeEmpty) {
-          if (value == null || value.isEmpty) {
+        if (value == null || value.isEmpty) {
+          if (canBeEmpty) {
             return null;
+          } else {
+            return noValueError;
           }
-          validator?.call(value);
-          return null;
         }
-        if (value == null || value.isEmpty) {
-          return noValueError;
+
+        if (isInt) {
+          final intValue = int.tryParse(value);
+          if (intValue == null || intValue < 0) {
+            return TextConstants.invalidNumber;
+          }
         }
-        final intValue = int.tryParse(value);
-        if (isInt && (intValue == null || intValue < 0)) {
-          return TextConstants.invalidNumber;
+
+        if (isDouble) {
+          final doubleValue = double.tryParse(value.replaceAll(',', '.'));
+          if (doubleValue == null || doubleValue < 0) {
+            return TextConstants.invalidNumber;
+          }
         }
-        final doubleValue = double.tryParse(value.replaceAll(',', '.'));
-        if (isDouble && (doubleValue == null || doubleValue < 0)) {
-          return TextConstants.invalidNumber;
+
+        if (validator == null) {
+          return null;
+        } else {
+          return validator!(value);
```

Comment:
Go for `validator?.call(value);`, it does the same thing as those four lines

Reply:
It doesn't. You need these two lines to achieve the same thing.
```
validator?.call(value);
return null;
```
I just thought that would make it easier to understand with the 4 lines.